File: gateway/src/main/java/hu/bme/aut/classifiedadvertisementsite/gateway/controller/exception/InternalServerErrorException.java
package hu.bme.aut.classifiedadvertisementsite.gateway.controller.exception;

import org.springframework.http.HttpStatus;

public class InternalServerErrorException extends AbstractException {
    public InternalServerErrorException(String errorMessage) {
        super(errorMessage);
    }

    @Override
    public HttpStatus getHttpStatus() {
        return HttpStatus.INTERNAL_SERVER_ERROR;
    }
}

File: advertisementservice/src/main/kotlin/hu/bme/aut/classifiedadvertisementsite/advertisementservice/controller/exception/InternalServerErrorException.kt
package hu.bme.aut.classifiedadvertisementsite.advertisementservice.controller.exception

import org.springframework.http.HttpStatus

class InternalServerErrorException(errorMessage: String) : AbstractException(errorMessage) {
    override fun getHttpStatus(): HttpStatus {
        return HttpStatus.INTERNAL_SERVER_ERROR
    }
}